### PR TITLE
Updated access.swift plugin

### PR DESCRIPTION
**Pull request by FireFoxIXI**
- Added rename function to swift wrapper
- Added directory stats handling to driver
- Added per user bucket creation to driver
- Added custom PSR4 loader as the composer loader conflicts with the composer loader in sabredav when using WebDAV

### CODE REVIEW

File: core/src/plugins/access.swift/class.swiftAccessDriver.php [hunk 1]
```diff
@@ -125,4 +155,10 @@ public function makeSharedRepositoryOptions($httpVars, $repository)
         return $newOptions;
     }
 
+    public function directoryUsage($directoryPath, $repositoryResolvedOptions)
```
Comment (cdujeu) [inline core/src/plugins/access.swift/class.swiftAccessDriver.php]: Can you add a note that this implements only the top-level bucket bytesize, without taking into account the directoryPath?

Comment (FireFoxIXI) [inline core/src/plugins/access.swift/class.swiftAccessDriver.php]: This returns the top-level bucket bytesize and ignores the directoryPath.

Comment (FireFoxIXI) [inline core/src/plugins/access.swift/class.swiftAccessDriver.php]: Bonjour Charles,

added that note,

Best regards,

Steve

---
Steve Ludovicy

Global Media Systems SA
212, route de Thionville -   L-2610 Howald
tel. +352 48 28 11   -   fax +352 48 28 110

Comment (cdujeu) [inline core/src/plugins/access.swift/class.swiftAccessDriver.php]: I meant adding a comment in the code, and update the PR :-)

**Comment by cdujeu** [comment]
steve, last small thing: can you make sure that your latest code does not create a blocking error if the swift sdk is actually not present ? Something like removing the openstack-sdk and trying to enter the workspace, how does it behave?
thx

**Comment by cdujeu** [comment]
@FireFoxIXI could you check? I'm afraid the code in "performChecks()" function may break everything if the sdk is not there.

**Comment by FireFoxIXI** [comment]
Hello,

I tested this, when the sdk is not present under

Application Parameters > Feature plugins  >Workspaces Drivers

OpenStack (Swift) under Automatic checks it then says:

"You must download the openstack-sdk-php and install it with Composer 
for this plugin"

This is the same behaviour as the access.s3 plugin for example.

Is there anything more that should be done there?

Best regards,

Steve

---
Steve Ludovicy

Global Media Systems SA
212, route de Thionville -   L-2610 Howald
tel. +352 48 28 11   -   fax +352 48 28 110

**Comment by cdujeu** [comment]
Now deprecated

**Pull request closed without merging**